geniusyield-market-maker/src/GeniusYield/MarketMaker/Prices.hs

```diff
@@ -539,7 +540,11 @@ data PricesProviderException
   | PPTaptoolsErr TaptoolsPriceException
   | PPMockErr MockPriceException
   deriving stock Show
-  deriving anyclass Exception
+
+instance Exception PricesProviderException where
+  displayException (PPMaestroErr err)  = "Maestro fail: " ++ displayException err
```

Comment:
Using default `Show` instance for `displayException` would make it clear when investigating logs which exception case is logged. `PPMaestroErr` gives clear indication of raised exception type, v/s string `Maestro fail: `.

Reply:
Assume for a moment that we agree to settle on the idea of modifying `displayException`.  Then this custom instance for `PricesProviderException` is necessary to pass the responsibility of displaying the error to `TaptoolsPriceException`.  The modifications to the "Maestro" and "Mock" cases are there just to keep the symmetry.